cai_causal_graph/graph_components.py

```diff
@@ -442,6 +498,33 @@ def details(self) -> str:
         """Return a detailed string description of the object."""
         return self.__repr__()
 
+    @classmethod
+    def from_dict(cls, edge_dict: dict) -> Edge:
+        """
+        Deserialize a dictionary to a `cai_causal_graph.graph_components.Edge` instance.
+
+        :param edge_dict: The dictionary representation of the `cai_causal_graph.graph_components.Edge` instance.
+        :return: The `cai_causal_graph.graph_components.Edge` instance.
+        """
+        source_node_class = edge_dict['source'].get('node_class', 'Node')
```

Comment:
Include the comment I provided in sample code so it is obvious why we are doing this. Need to point out that we will default to Node if not provided.